### DISCUSSION TITLE
OTel Demo App | Bring your own backend | Elastic Search

### DISCUSSION THREAD

**Discussion by Vyankatesh5** [open-telemetry/opentelemetry-demo#884] [Q&A] [answered]
I am trying to send the trace and metric of demo to elk stack. Edited the configurations as below ,
Configuration of 
`[src/otelcollector/otelcol-config-extras.yml](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/otelcollector/otelcol-config-extras.yml)`
```exporters:
  elasticsearch/trace:
    endpoints: [https://elasticIP:9200]
    traces_index: trace_index
    user: Username
    password: Password
    tls:
      insecure_skip_verify: true
  elasticsearch/log:
    endpoints: [http://elasticIP:9200]
    logs_index: my_log_index
    sending_queue:
      enabled: true
      num_consumers: 20
      queue_size: 1000

service:
  pipelines:
    logs:
      receivers: [otlp]
      processors: [batch]
      exporters: [elasticsearch/log]
    traces:
      receivers: [otlp]
      exporters: [elasticsearch/trace]
      processors: [batch]    
```

But the container `otel/opentelemetry-collector-contrib:0.76.1` is failing to start 
error 
`Error: invalid configuration: connectors::spanmetrics: must be used as both receiver and exporter but is not used as exporter
2023/05/10 09:17:47 collector server run finished with error: invalid configuration: connectors::spanmetrics: must be used as both receiver and exporter but is not used as exporter`
Can anyone help me to integrate this with elastic?

**Comment by julianocosta89** [comment, marked as answer]
This problem was discussed here:
https://github.com/open-telemetry/opentelemetry-demo/discussions/847

The issue here is that your service pipelines are overwriting the `otelcol-config.yaml`.

Could you try with the following:

```
service:
  pipelines:
    traces:
      exporters: [otlp, logging, spanmetrics,elasticsearch/trace]
    logs:
      exporters: [logging,elasticsearch/log]
```
